File: client/react/hrms/src/services/candidateService.js
import axios from "axios";

export default class CandidateService {
  getAll() {
    return axios.get("http://localhost:8080/api/candidates/getall");
  }

  getById(id) {
    return axios.get("http://localhost:8080/api/candidates/getbyid?id=" + id);
  }

  save(candidate) {
    return axios.post("http://localhost:8080/api/candidates/save", candidate);
  }
}

File: server/java/src/main/java/com/hrms/business/abstracts/CandidateService.java
package com.hrms.business.abstracts;

import com.hrms.core.utilities.results.DataResult;
import com.hrms.core.utilities.results.Result;
import com.hrms.entities.concretes.Candidate;
import com.hrms.entities.dtos.CandidateResumeDto;

import java.util.List;

public interface CandidateService {

    Result save(Candidate candidates);
    DataResult<Candidate> getByIdentificationNumber(String identificationNumber);
    DataResult<Candidate> getById(int id);
    DataResult<List<Candidate>> getAll();
    DataResult<CandidateResumeDto> getResumeByCandidateId(int candidateId);

}
